### PR TITLE
fix(cli/console): escape non printable characters in object entries

### LINKED ISSUE #7532: Inspect should escape terminal sequences on object keys
For example, given an array like the following

```typescript
const values = [
  "\x1b[32mhello\x1b[39m",
  "\x1b[2J",
];

console.log(Deno.inspect(values));
```

I'd expect to see output along the lines of:

```
[
  "\\x1b[32mhello\\x1b[39m",
  "\\x1b[2J",
]
```

Instead we get a escape sequences printed as is which in turn gets interpreted by the terminal, in the given example it would be a garbled screen because of the clear sequence.

### PR DESCRIPTION
**Pull request by caspervonb**
Fixes #7532

### CODE REVIEW

**Comment by solson** [comment]
Would it make sense to replace *all* other non-printable characters with hex escape codes?

**Comment by caspervonb** [comment]
> Would it make sense to replace all other non-printable characters with hex escape codes

Hmm, probably, Node's inspect does this too.

**Comment by solson** [comment]
Something like this should do the trick (placed after the others, as a catch-all):

```js
.replace(/[\x00-\x1f]/g, c => "\\x" + c.charCodeAt(0).toString(16).padStart(2, '0'))
```

File: cli/rt/02_console.js [hunk 1]
```diff
@@ -462,7 +462,11 @@
       .replace(/\n/g, "\\n")
       .replace(/\r/g, "\\r")
       .replace(/\t/g, "\\t")
-      .replace(/\v/g, "\\v");
+      .replace(/\v/g, "\\v")
+      .replace(
+        /[\x00-\x1f]/g,
```
Comment (nayeemrmn) [inline cli/rt/02_console.js:467]:
```suggestion
        /[\x00-\x1f\x7f-\x99]/g,
```

Comment (caspervonb) [reply cli/rt/02_console.js]: Oh yeah DEL et cetera, thanks @nayeemrmn 👍

**Review by crypt096 — APPROVED** [review]

File: cli/rt/02_console.js (hunk 1 above)
Comment (solson) [reply cli/rt/02_console.js]: It looks like this should extend to `\x9f`: https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes

Comment (nayeemrmn) [reply cli/rt/02_console.js]: Oops, I went from `\x99` to `\u0100` and didn't check.

File: cli/rt/02_console.js [hunk 2]
```diff
@@ -462,7 +462,11 @@
       .replace(/\n/g, "\\n")
       .replace(/\r/g, "\\r")
       .replace(/\t/g, "\\t")
-      .replace(/\v/g, "\\v");
+      .replace(/\v/g, "\\v")
+      .replace(
+        /[\x00-\x1f\x7f-\x99]/g,
```
Comment (solson) [inline cli/rt/02_console.js:467]:
```suggestion
        /[\x00-\x1f\x7f-\x9f]/g,
```

It looks like this should extend to `\x9f`: https://en.wikipedia.org/wiki/List_of_Unicode_characters#Control_codes

**Review by ry — APPROVED** [review]
LGTM - thanks Casper

**Pull request merged**